src/components/timeline/Timeline.svelte

```diff
@@ -21,35 +43,125 @@
 
   let clientWidth: number = 0;
   let cursorEnabled: boolean = true;
-  let cursorHeaderHeight: number = 20;
+  let cursorHeaderHeight: number = 0;
+  // let cursorHeaderHeight: number = 20;
   let histogramCursorTime: Date | null = null;
   let mouseOver: MouseOver;
   let mouseOverViolations: MouseOverViolations;
   let rowDragMoveDisabled = true;
   let rowsMaxHeight: number = 600;
   let rows: Row[] = [];
-  let tickCount: number = 5;
+  let tickCount: number = 20;
   let timeline: Timeline;
   let timelineDiv: HTMLDivElement;
   let timelineHistogramDiv: HTMLDivElement;
   let timelineHistogramDrawHeight: number = 40;
   let xAxisDiv: HTMLDivElement;
-  let xAxisDrawHeight: number = 56;
+  let xAxisDrawHeight: number = 64;
+  let xTicksView = [];
 
   $: timeline = $view?.definition.plan.timelines.find(timeline => timeline.id === timelineId);
   $: rows = timeline?.rows || [];
   $: timeline.gridId = gridId;
   $: drawWidth = clientWidth > 0 ? clientWidth - timeline?.marginLeft - timeline?.marginRight : 0;
+
+  $: if (drawWidth) {
+    let ticks = Math.round(drawWidth / 140);
+    tickCount = clamp(ticks, 2, 15);
+  }
+
   $: setRowsMaxHeight(timelineDiv, xAxisDiv, timelineHistogramDiv);
   $: xDomainMax = [new Date($maxTimeRange.start), new Date($maxTimeRange.end)];
   $: xDomainView = [new Date($viewTimeRange.start), new Date($viewTimeRange.end)];
   $: xScaleMax = getXScale(xDomainMax, drawWidth);
   $: xScaleView = getXScale(xDomainView, drawWidth);
-  $: xTicksView = xScaleView.ticks(tickCount).map((date: Date) => {
-    const doyTimestamp = getDoyTime(date, false);
-    const [yearDay, time] = doyTimestamp.split('T');
-    return { date, time, yearDay };
-  });
+  $: xScaleViewDuration = $viewTimeRange.end - $viewTimeRange.start;
+
+  // Use a custom D3 time day to force equidistant time intervals
+  // for days as opposed to D3's non-uniform intervals that can end early
+  // on months or years
+  // See https://github.com/d3/d3-scale/issues/245
+  // And https://observablehq.com/d/906f777c9f2f0701
+  const customD3TimeDay = timeInterval(
```

Comment:
Are we able to move this to `utilities/timeline.ts`?

Reply:
Yep